simulations/src/node/carnot/researsh_overlay.rs

```diff
@@ -0,0 +1,98 @@
+use consensus_engine::{overlay::LeaderSelection, Committee, NodeId, TimeoutQc};
+
+use crate::node::CommitteeExt;
+
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
+pub struct ResearchOverlaySettings<L: LeaderSelection> {
+    num_nodes: usize,
+    leader: L,
+    leader_threshold: usize,
+}
+
+#[derive(Debug, Clone)]
+pub struct ResearchOverlay<L: LeaderSelection> {
+    committee: Committee,
+    nodes: Vec<NodeId>,
+    l: L,
+    leader_threshold: usize,
+}
+
+impl<L: LeaderSelection + Send + Sync + 'static> consensus_engine::Overlay for ResearchOverlay<L> {
+    type Settings = ResearchOverlaySettings<L>;
+
+    type LeaderSelection = L;
+
+    fn new(settings: Self::Settings) -> Self {
+        let mut c = Committee::new();
+        c.fill(settings.num_nodes, 0, None);
+        Self {
+            nodes: c.iter().copied().collect(),
+            committee: c,
+            l: settings.leader,
+            leader_threshold: settings.leader_threshold,
+        }
+    }
+
+    fn rebuild(&mut self, _timeout_qc: TimeoutQc) {}
+
+    fn is_member_of_child_committee(&self, _parent: NodeId, _child: NodeId) -> bool {
+        false
+    }
+
+    fn is_member_of_root_committee(&self, _id: NodeId) -> bool {
+        true
+    }
+
+    fn is_member_of_leaf_committee(&self, _id: NodeId) -> bool {
+        true
+    }
+
+    fn is_child_of_root_committee(&self, _id: NodeId) -> bool {
+        false
+    }
+
+    fn parent_committee(&self, _id: NodeId) -> Committee {
+        std::iter::once(self.next_leader()).collect()
+    }
```

Comment:
Definetely something wrong here. As the overlay trait changed this method signature.